src/weights.jl

```diff
@@ -250,6 +250,76 @@ eweights(t::AbstractVector, r::AbstractRange, λ::Real) =
 
 # NOTE: No variance correction is implemented for exponential weights
 
+struct UnitWeights{S<:Real, T<:Real} <: AbstractWeights{S, T, V where V<:Vector{T}}
+    sum::S
+end
+
+UnitWeights{T}(s::S) where {S, T} = UnitWeights{S, T}(s)
+
+@doc """
+    UnitWeights{T}(s)
+
+Construct a `UnitWeights` vector with length `s` and weight elements of type `T`.
+All weight elements are identically one.
+""" UnitWeights
+
+eltype(wv::UnitWeights{S, T})    where {S, T} = T
+values(wv::UnitWeights{S, T})    where {S, T} = fill(one(T), length(wv))
+isempty(wv::UnitWeights{S, T})   where {S, T} = iszero(wv.sum)
+length(wv::UnitWeights{S, T})    where {S, T} = Int(wv.sum)
+Base.size(wv::UnitWeights{S, T}) where {S, T} = Tuple(length(wv))
+
+function Base.getindex(wv::UnitWeights{S, T}, i::Int) where {S, T}
+    (i > 0) & (i <= length(wv)) ? one(T) : Base.throw_boundserror(wv, i)
+end
+
+function Base.getindex(wv::UnitWeights{S, T}, i::AbstractRange{<:Int}) where {S, T}
+    if (minimum(i) > 0) & (maximum(i) <= length(wv))
+        return fill(one(T), length(i))
+    else
+        Base.throw_boundserror(wv, i)
+    end
+end
+
+function Base.getindex(wv::UnitWeights{S, T}, i::AbstractArray{<:Int}) where {S, T}
+    if (minimum(i) > 0) & (maximum(i) <= length(wv))
+        return fill(one(T), size(i))
+    else
+        Base.throw_boundserror(wv, i)
+    end
+end
+
+Base.getindex(wv::UnitWeights{S, T}, ::Colon) where {S, T} = fill(one(T), length(wv))
+
+"""
+    uweights(::Type{T}, s::Real) where T<:Real
+    uweights(s::T) where T<:Real
+
+Construct a `UnitWeights` vector with length `s` and weight elements of type `T`.
+All weight elements are identically one.
+
+# Examples
+```julia-repl
+julia> uweights(Float64, 3)
+3-element UnitWeights{Int64,Float64}:
+ 1.0
+ 1.0
+ 1.0
+```
+"""
+uweights(::Type{T}, s::S) where {S, T} = UnitWeights{S, T}(s)
+uweights(s::S)            where S      = UnitWeights{S, S}(s)
+
+"""
+    varcorrection(w::UnitWeights, corrected=false)
+
+* `corrected=true`: ``\\frac{1}{\\sum w - \\sum {w^2} / \\sum w}``
```

Comment:
The formula doesn't reflect the code. Though I would expect all three formulas for frequency, probability and analytical weights to give the same result for this special case, right? If so, probably worth mentioning.

Reply:
Sorry, mindless copying and pasting